src/Command/Ingester/CreateUserCommand.php

```diff
@@ -0,0 +1,461 @@
+    }
+
+    private function initializeLineItemSlugsOption(string $lineItemSlugs): void
+    {
+        if (!empty($lineItemSlugs)) {
+            $this->lineItemSlugs = array_filter(
+                explode(',', (string)$lineItemSlugs),
+                static function (string $value): bool {
+                    return !empty($value);
+                }
+            );
+        }
+        if (empty($this->lineItemSlugs)) {
+            throw new InvalidArgumentException(
+                sprintf("Invalid '%s' option received.", self::OPTION_LINE_ITEM_SLUGS)
+            );
+        }
+    }
+
+    private function initializeUserPrefixOption(InputInterface $input): void
+    {
+        $this->userPrefix = array_filter(
+            explode(',', (string)$input->getArgument('user-prefix')),
+            static function (string $value): bool {
+                return !empty($value);
+            }
+        );
+
+        if (empty($this->userPrefix)) {
+            throw new InvalidArgumentException(
+                sprintf("User Prefix is a required argument")
+            );
+        }
+    }
+
+    private function createUserPassword(): string
+    {
+        return substr(str_shuffle("0123456789abcdefghijklmnopqrstvwxyz"), 0, 8);
+    }
+
+    /**
+     *
+     * Function to create user data
+     */
+    private function createUserDto(string $username, string $userPassword, string $userGroupId): UserDto
+    {
+        return new UserDto(
+            $username,
+            $this->passwordEncoder->encodePassword(new User(), $userPassword),
+            $userGroupId != '' ?  $userGroupId : null
+        );
+    }
+
+    /**
+     * Function to create user assignment data
+     */
+    private function createAssignmentDto(
+        LineItemCollection $lineItems,
+        string $lineItemSlug,
+        string $username
+    ): AssignmentDto {
+        $lineItem = $lineItems->getBySlug($lineItemSlug);
+        return new AssignmentDto(Assignment::STATE_READY, (int)$lineItem->getId(), $username);
+    }
+
+    private function getFindLineItemCriteria(): FindLineItemCriteria
+    {
+        $criteria = new FindLineItemCriteria();
+
+        if (!empty($this->lineItemIds)) {
+            $criteria->addLineItemIds(...$this->lineItemIds);
+        }
+        return $criteria;
+    }
+
+    /**
+     * @throws LineItemNotFoundException
+     */
+    private function getAllLineItemSlugs(): Void {
+        $lineItems = $this->lineItemRepository->findAllSlugsAsArray();
+        if (empty($lineItems)) {
+            throw new LineItemNotFoundException("No line items were found in database.");
+        }
+        $this->lineItemSlugs = array_column($lineItems, 'slug');
+    }
+
+    /**
+     * Function to check whether a user exist for a Line Item Slug and returns last increment no of it
+     * 
+     */
+    private function checkLineItemSlugUserExist(
+        LineItemCollection $lineItems,
+        Array $lineItemSlugs
+    ): Array {
```

Comment:
Please check the types here, it should be `array` and not `Array`

Reply:
Done